main.tf

```diff
@@ -1,13 +1,254 @@
-resource "random_integer" "example" {
-  count = module.this.enabled ? 1 : 0
+data "aws_caller_identity" "current" {}
+data "aws_region" "current" {}
```

Comment:
use `var.region` instead of searching for the current region.
This way, the caller of the module would decide in which region to provision the resources

Reply:
Updated